docs/core/examples/simpleclient.py

```diff
@@ -5,23 +5,39 @@
 """
 An example client. Run simpleserv.py first before running this.
 """
+from zope.interface import implementer
 
 from twisted.internet import reactor, protocol
+from twisted.internet.interfaces import IProtocol
 
 
 # a client protocol
 
 
-class EchoClient(protocol.Protocol):
+@implementer(IProtocol)
+class EchoClient:
     """Once connected, send a message, then print the result."""
 
+    def makeConnection(self, transport):
+        """
+        Called when a new connection is made.
+
+        C{transport} is used to send data to the remote peer.
+        """
+        self._transport = transport
+        print(f"New connection from remote peer {transport}")
+        self._transport.write(b"hello, world!")
+
     def connectionMade(self):
-        self.transport.write(b"hello, world!")
+        """
+        Part of IProtocol interface, but not used for non inheritance
+        base implementations.
+        """
```

Comment:
```suggestion
        """
        Part of IProtocol interface, but not used for non-inheritance
        based implementations.
        """
```

Reply:
This is a weird behavior for `IProtocol`… Wouldn't a better (at least more consistent) example be to have `makeConnection()` call `connectionMade()` and keep the `write()` in here?

Presumably, should this example class be subclassed, that subclass might try to implement `connectionMade()` based on the `IProtocol` docs and be a bit confused about why it never gets called.

That suggests to me that the `IProtocol` docs should say that `makeConnection()` is expected to call `connectionMade()`.